src/engine/sequencer.js

```diff
@@ -217,11 +216,9 @@ Sequencer.prototype.stepToProcedure = function (thread, procedureCode) {
         var doWarp = definitionBlock.mutation.warp;
         if (doWarp) {
             thread.peekStackFrame().warpMode = true;
-        } else {
+        } else if (isRecursive) {
             // In normal-mode threads, yield any time we have a recursive call.
-            if (isRecursive) {
-                thread.status = Thread.STATUS_YIELD;
-            }
+            thread.status = Thread.STATUS_YIELD;
```

Comment:
Another case where I think it's more clear to step into the else and then consider the if, rather than considering both at once. The `isRecursive` just seems like a different concern than `!doWarp`, so considering them together in the same line of logic is more confusing than helpful I think.